src/sentry/sentry_apps/token_exchange/grant_exchanger.py

```diff
@@ -108,18 +108,28 @@ def grant(self) -> ApiGrant:
                 .get(code=self.code)
             )
         except ApiGrant.DoesNotExist:
-            raise APIUnauthorized("Could not find grant")
+            raise SentryAppIntegratorError(
+                "Could not find grant for given code",
+                webhook_context={"code": self.code[:4]},
+                status_code=404,
```

Comment:
`APIUnauthorized` has status code 401.

```suggestion
                webhook_context={"code": self.code[:4]},
                status_code=401,
```

Reply:
if you were looking for object with id 123 and it doesn't exist i think it makes sense to raise 404. but if you are looking for an `APIGrant` with a code and you can't find one, it's a little less clear. i'm not particular about which one, but it seems like you made the change previously to return `ApiUnauthorized` with the message?

Reply:
yeah for these files I just defaulted to `APIUnauthorized` for all of them, which in retrospect is not ideal